Libraries/LibWeb/DOM/EventTarget.cpp

```diff
@@ -848,16 +859,22 @@ bool EventTarget::dispatch_event(Event& event)
         auto unsafe_shared_time = HighResolutionTime::unsafe_shared_current_time();
         auto current_time = HighResolutionTime::relative_high_resolution_time(unsafe_shared_time, realm().global_object());
 
-        if (is<HTML::Window>(this)) {
-            auto* window = static_cast<HTML::Window*>(this);
+        GC::Ptr<HTML::Window> window = [&]() {
```

Comment:
This fix for properly dispatching to Document's window seems like a separate fix. Can you split it into its own commit?

Reply:
Absolutely, can it be a part of this PR?